src/main/functions/DidSiopMetadata.ts

```diff
@@ -30,29 +32,50 @@ function getIntersection<T>(rpMetadata: Array<T> | T, opMetadata: Array<T> | T):
   return arrayA.filter((value) => arrayB.includes(value));
 }
 
-function verifySubjectIdentifiers(subjectIdentifiers: SubjectIdentifierType | SubjectIdentifierType[]): boolean {
-  if (subjectIdentifiers || subjectIdentifiers.length) {
-    if (Array.isArray(subjectIdentifiers)) {
-      return subjectIdentifiers.includes(SIOP.SubjectIdentifierType.DID);
+function verifySubjectIdentifiers(subjectSyntaxTypesSupported: string[]): boolean {
```

Comment:
Are you really sure that a single non array element isn't allowed anymore?

Reply:
yes, I haven't seen singular form anywhere

Reply:
If you haven't seen it, is there a normative text in the spec that makes clear it only can be an array. Otherwise we are now removing a more relaxed handling of the object. In other words, if we support both it will not hurt anyone.

Reply:
I'm not sure what you mean, but both [jwt-vc-presentation-profile](https://identity.foundation/jwt-vc-presentation-profile/#verifierrp-registration-metadata) and [SIOPv2](https://openid.net/specs/openid-connect-self-issued-v2-1_0-06.html#section-9.2-5.8.1) are only talking about jsonarrays

Reply:
The first spec is only something on top of the underlying specs, and should never be taken into account in a lib which can also be used outside of the interop profile. 

The second does make it clear that it needs to be an array. But since there is no schema yet, I would rather have us be relaxed about it. So whenever the arguments is being set from a developer we expect an array to be passed in. But of it comes in as a payload we can accept a single string instead of an array.

Reply:
changed all of the list types to accept a singular value as well